whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticFullTextIndex.java

```diff
@@ -67,6 +68,38 @@ public void scheduledUpdate() {
         LOGGER.info("Completed updating Elasticsearch indexes");
     }
 
+    @Scheduled(fixedDelay = 60 * 60 * 1_000L)
+    @SchedulerLock(name = ALERT_TASK_NAME)
+    public void checkIndex() {
+        if (!elasticIndexService.isEnabled()) {
+            LOGGER.error("Elasticsearch is not enabled");
+            return;
+        }
+
+        try {
+            alert();
+        } catch (DataAccessException | IOException | IllegalStateException e) {
+            LOGGER.error("Unable to perform fulltext index checks due to {}: {}", e.getClass(), e.getMessage());
+        }
+
+        LOGGER.info("Completed Elasticsearch index check");
+    }
+
+    protected void alert() throws IOException{
+        if (shouldRebuild()) {
+            LOGGER.error("ES indexes needs to be rebuild");
+            return;
+        }
+
+        final Map<Integer, Integer> maxSerialIdWithObjectCount = serialDao.getMaxSerialIdWithObjectCount();
+        final int countInDb = ((int) maxSerialIdWithObjectCount.values().toArray()[0]);
+        final long countInES = elasticIndexService.getWhoisDocCount();
+        if (countInES != countInDb) {
+            final int dbMaxSerialId = (Integer) maxSerialIdWithObjectCount.keySet().toArray()[0];
+            LOGGER.error(String.format("Number of objects in DB (%s) does not match to number of objects indexed in ES (%s) for serialId (%s)", countInDb, countInES, dbMaxSerialId));
```

Comment:
Don't you need to call refresh now? Else we will have same issue?

Reply:
I need to take a coffee, yes, of course that I need to call refresh